### PR TITLE
Issue/8303 document environment variables for config options

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

Documentation PR for https://github.com/inmanta/inmanta-core/issues/8303

closes  https://github.com/inmanta/inmanta-core/issues/8303

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

**Comment by Hugo-Inmanta** [comment]
Latest look: 
image

File: docs/reference/config.rst [hunk 1]
```diff
@@ -3,9 +3,36 @@
...
+
+.. code-block:: bash
+
+    export INMANTA_DATABASE_CONNECTION_TIMEOUT=60
+
+If an option is set both via a config file and via an environment variable,
+the environment variable value will take precedence.
+
+For more information about how to use the configuration framework and details
+about precedence rules, please visit the administrator documentation configuration :ref:`page<configuration_framework>`.
```
Comment (Hugo-Inmanta) [inline docs/reference/config.rst:32]: Some of it is redundant with the administrator documentation page, and I'm not sure how much value the example brings. Maybe only add the link to the administrator documentation page ?

**Comment by Hugo-Inmanta** [comment]
@arnaudsjs I'm not sure about the relative pinning between core and inmanta-sphinx. Currently tests for [inmanta-sphinx](https://github.com/inmanta/inmanta-sphinx/pull/79) are failing because the code is not there present on core master.
And the tests here are failing because inmanta-sphinx 2.2.0 that contains required changed is not released yet.

File: tests/test_config.py [hunk 2]
```diff
@@ -51,6 +51,17 @@ def test_options(monkeypatch):
 
 
 def test_configfile_hierarchy(monkeypatch, tmpdir):
+    """
+    Test the hierarchy when a config option is set in multiple places:
+
+        - in the default main config file /etc/inmanta/inmanta.cfg
+        - in a config directory --config-dir cli option
+        - in a config file passed via --config cli option
+        - via environment variable
```
Comment (arnaudsjs) [inline tests/test_config.py:60]: This doesn't mentioned to dot files.

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

**Comment by arnaudsjs** [comment]
> @arnaudsjs I'm not sure about the relative pinning between core and inmanta-sphinx. Currently tests for [inmanta-sphinx](https://github.com/inmanta/inmanta-sphinx/pull/79) are failing because the code is not there present on core master. And the tests here are failing because inmanta-sphinx 2.2.0 that contains required changed is not released yet.

Ideally you do this via three PRs:
* One PR on inmanta-core that adds the `get_environment_variable()` method.
* Then the PR on inmanta-sphinx.
* Finally the PR on inmanta-core that updates the documentation. (This PR can bump the version of inmanta-sphinx, but it might be that dependabot is quicker.)

**Review by arnaudsjs — APPROVED** [review]

**Pull request closed without merging**